src/api.rs

```diff
@@ -891,16 +952,26 @@ impl<T: Pixel> ContextInner<T> {
     }
   }
 
-  pub fn send_frame<F>(&mut self, frame: F) -> Result<(), EncoderStatus>
-  where
-    F: Into<Option<Arc<Frame<T>>>>
+  pub fn send_frame(&mut self,
+                    frame: Option<Arc<Frame<T>>>,
+                    props: Option<FrameParameters>) -> Result<(), EncoderStatus>
   {
     let input_frameno = self.frame_count;
-    let frame = frame.into();
     if frame.is_some() {
       self.frame_count += 1;
     }
     self.frame_q.insert(input_frameno, frame);
+
+    if let Some(props) = props {
+      if props.keyframe {
+        self.keyframes_forced.insert(input_frameno);
+      }
+    }
+
+    if !self.needs_more_lookahead() {
+      self.compute_lookahead_data();
```

Comment:
Is this related to or on top of the other PR (lookahead into send_frame)?

Reply:
It is not related, there should be another pr open I can move this away but I needed to make sure it works with it.

Reply:
Well, as long as it's not misleading or broken...